py_gtfs_rt_ingestion/ingest.py

```diff
@@ -83,26 +83,52 @@ def lambda_handler(event: dict, context) -> None:
     batch files should all be of same ConfigType
     """
     logging.info("Processing event:\n%s" % json.dumps(event, indent=2))
-    temp_dir = None
-    if 'files' not in event:
-        temp_dir = tempfile.TemporaryDirectory()
-        files = [download_file_from_s3(event['bucket'], file, temp_dir.name)
-                 for file in event['s3_files']]
 
-    else:
+    # get files and function to read them based on the event. for local files,
+    # use gzip reading, for s3 files use pyarrow to read directly from s3
+    if 'files' in event:
         files = event['files']
+        conversion_func = gz_to_pyarrow
+    elif 's3_files' in event:
+        files = event['s3_files']
+        conversion_func = s3_to_pyarrow
+    else:
+        raise Exception("poorly formatted event")
```

Comment:
figure out a conversion func based on the keys filled out in the event.

throw an exception otherwise. if an event is poorly formatted we should automatically cancel the job.